designs/2024-baseline-support/README.md

```diff
@@ -47,161 +47,191 @@ The baseline files includes details about the file where the error found, the ru
 }
 ```
 
-### Generating the baseline
+The file is stored in `.eslint-suppressions.json` , unless otherwise specified.
 
-A new option `--baseline` wil be introduced to ESLint CLI. When provided, the baseline is generated and saved in `.eslint-baseline.json`. If the file already exists, it gets over-written. Note that this is a boolean flag option (no values are accepted). Here is an example how to generate the baseline:
+### Suppressing all violations
+
+A new option `--suppress-all` wil be introduced to ESLint CLI. When provided, the JSON file is generated and saved in `.eslint-suppressions.json`. If the file already exists, it gets over-written. Note that this is a boolean flag option (no values are accepted).
 
 ``` bash
-eslint --baseline ./src
+eslint --suppress-all ./src
 ```
 
-The above goes through each result item and messages, and counts the number of errors (`severity == 2`). The necessary details are stored in the baseline file. By default, the baseline file is saved at `.eslint-baseline.json` . To control where the baseline is saved, another option will be introduced `--baseline-location`. That is a string argument specifying where the baseline must be saved. Here is an example 
-how to generate the baseline and save the results to `/home/user/project/mycache`.
+### Suppressing violations of a specific rule
+
+A new option `--suppress-rule [RULE1]` will be introduced to ESLint CLI. When provided, the existing suppressions file will be updated to include any existing violation of the provided rule. The suppressions file will be created if not already exists. Note that this is a string flag option (value is required).
 
 ``` bash
-eslint --baseline --baseline-location /home/user/project/mycache
+eslint --suppress-rule '@typescript-eslint/no-explicit-any' ./src
```

Comment:
I think we should allow `--suppress-rule` to be passed multiple times to specify multiple rules in one run.